Formula/petsc.rb

```diff
@@ -0,0 +1,197 @@
+  depends_on "hwloc"        => :recommended
+  depends_on "suite-sparse" => :recommended
+  depends_on "netcdf"       => :recommended
+  depends_on "fftw"         => ["with-mpi", :recommended]
+
+  def oprefix(f)
+    Formula[f].opt_prefix
+  end
+
+  def install
+    ENV.deparallelize
+
+    arch_real="real"
+    arch_complex="complex"
+
+    # Environment variables CC, CXX, etc. will be ignored by PETSc
+    ENV.delete "CC"
+    ENV.delete "CXX"
+    ENV.delete "F77"
+    ENV.delete "FC"
+    # PETSc is not threadsafe, disable pthread/openmp (see http://www.mcs.anl.gov/petsc/miscellaneous/petscthreads.html)
+    args = %w[CC=mpicc
+              CXX=mpicxx
+              F77=mpif77
+              FC=mpif90
+              --with-shared-libraries=1
+              --with-pthread=0
+              --with-openmp=0]
+    args << ("--with-debugging=" + ((build.with? "debug") ? "1" : "0"))
+
+    # We don't download anything, so no need to build against openssl
+    args << "--with-ssl=0"
+
+    if build.with? "superlu43"
+      slu = Formula["superlu43"]
+      args << "--with-superlu-include=#{slu.include}/superlu"
+      args << "--with-superlu-lib=-L#{slu.lib} -lsuperlu"
+    end
+
+    args << "--with-fftw-dir=#{oprefix("fftw")}" if build.with? "fftw"
+    args << "--with-netcdf-dir=#{oprefix("netcdf")}" if build.with? "netcdf"
+    args << "--with-suitesparse-dir=#{oprefix("suite-sparse")}" if build.with? "suite-sparse"
+    args << "--with-hdf5-dir=#{oprefix("hdf5")}" if build.with? "hdf5"
+    args << "--with-metis-dir=#{oprefix("metis")}" if build.with? "metis"
+    args << "--with-scalapack-dir=#{oprefix("scalapack")}" if build.with? "scalapack"
+    args << "--with-x=0" if build.without? "x11"
+
+    # configure fails if those vars are set differently.
+    ENV["PETSC_DIR"] = Dir.getwd
+
+    # real-valued case:
+    ENV["PETSC_ARCH"] = arch_real
+    args_real = ["--prefix=#{prefix}/#{arch_real}", "--with-scalar-type=real"]
+    # TODO: compile separately (https://bitbucket.org/petsc/pkg-ml/commits/tag/v6.2-p3)
+    # --with-ml-include=/path/to/ml/include --with-ml-lib=/path/to/ml/liblibml.a
+    args_real << "--download-ml=1" if build.with? "ml"
+    args_real << "--with-hypre-dir=#{oprefix("hypre")}" if build.with? "hypre"
+    args_real << "--with-hwloc-dir=#{oprefix("hwloc")}" if build.with? "hwloc"
+    system "./configure", *(args + args_real)
+    system "make", "all"
+    system "make", "test" if build.with? "test"
+    system "make", "install"
+
+    # complex-valued case:
+    ENV["PETSC_ARCH"] = arch_complex
+    args_cmplx = ["--prefix=#{prefix}/#{arch_complex}", "--with-scalar-type=complex"]
+    system "./configure", *(args + args_cmplx)
+    system "make", "all"
+    system "make", "test" if build.with? "test"
+    system "make", "install"
+
+    # Link only what we want
+    petsc_arch = ((build.with? "complex") ? arch_complex : arch_real)
+
+    include.install_symlink Dir["#{prefix}/#{petsc_arch}/include/*h"],
```

Comment:
This should not be needed. If it is, then it should be done by `make install`.

Reply:
If we configure PETSc for real numbers only, then the build products could go directly into `#{prefix}`. One concern is that PETSc produces some non-executables in `#{prefix}/bin` that `brew --audit` complains about. Is there a standard procedure for this case?

Are there analytics on how often `petsc --with-complex` was used?

Reply:
@ilovezfs Building PETSc in prefix does not work "out of the box" because PETSc produces non-executable outputs in bin, which `brew audit` flags.

I see two options to work around this: build PETSc in `prefix` and somehow exclude the problematic files from Homebrew (perhaps ignore the bin dir, or `rm` the offending files?), or  build in a subdir and set up links to include, lib, etc (as was done here, but hopefully in a compliant way).

What is the preferred approach? Perhaps a third option?